backend/src/schema/resolvers/follow.spec.js

```diff
@@ -1,36 +1,93 @@
-import { GraphQLClient } from 'graphql-request'
+import { createTestClient } from 'apollo-server-testing'
 import Factory from '../../seed/factories'
-import { host, login } from '../../jest/helpers'
+import { getDriver } from '../../bootstrap/neo4j'
+import createServer from '../../server'
+import { gql } from '../../jest/helpers'
 
 const factory = Factory()
-let clientUser1
-let headersUser1
+const driver = getDriver()
 
-const mutationFollowUser = id => `
-  mutation {
-    follow(id: "${id}", type: User)
+let query
+let mutate
+let authenticatedUser
+
+let user1
+let user2
+let variables
+
+const mutationFollowUser = gql`
+  mutation($id: ID!, $type: FollowTypeEnum) {
+    follow(id: $id, type: $type) {
+      name
+      followedBy {
+        id
+        name
+      }
+      followedByCurrentUser
+    }
   }
 `
-const mutationUnfollowUser = id => `
-  mutation {
-    unfollow(id: "${id}", type: User)
+
+const mutationUnfollowUser = gql`
+  mutation($id: ID!, $type: FollowTypeEnum) {
+    unfollow(id: $id, type: $type) {
+      name
+      followedBy {
+        id
+        name
+      }
+      followedByCurrentUser
+    }
   }
 `
 
-beforeEach(async () => {
-  await factory.create('User', {
-    id: 'u1',
-    email: 'test@example.org',
-    password: '1234',
-  })
-  await factory.create('User', {
-    id: 'u2',
-    email: 'test2@example.org',
-    password: '1234',
+const userQuery = gql`
+  query($id: ID) {
+    User(id: $id) {
+      followedBy {
+        id
+      }
+      followedByCurrentUser
+    }
+  }
+`
+
+beforeAll(() => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-16T19:36:26Z" title="Monday, September 16th 2019, 9:36:26 pm +02:00">Sep 16, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -17,20 +24,42 @@ const mutationUnfollowUser = id => `
   }
 `
 
-beforeEach(async () => {
-  await factory.create('User', {
-    id: 'u1',
-    email: 'test@example.org',
-    password: '1234',
-  })
-  await factory.create('User', {
-    id: 'u2',
-    email: 'test2@example.org',
-    password: '1234',
+beforeAll(() => {
```


FYI @vbelolapotkov the `beforeAll` in most of our backend tests is *not only* a performance thing. There is also a bug in [graphql middlewares](https://github.com/prisma/graphql-middleware/issues/63#issuecomment-520786385) which will wrap another layer of middlewares around each resolver for every test case if you use `beforeEach`.

Reply:
> [<img alt="vbelolapotkov" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/vbelolapotkov) **Authored by [vbelolapotkov](https://github.com/vbelolapotkov)**
_<time datetime="2019-09-17T07:01:21Z" title="Tuesday, September 17th 2019, 9:01:21 am +02:00">Sep 17, 2019</time>_

---


Good point, thanks for the info